projects/RabbitMQ.Client/client/impl/RecordedConsumer.cs

```diff
@@ -29,16 +29,41 @@
 //  Copyright (c) 2007-2020 VMware, Inc.  All rights reserved.
 //---------------------------------------------------------------------------
 
+using System;
 using System.Collections.Generic;
 
 namespace RabbitMQ.Client.Impl
 {
     internal class RecordedConsumer : IRecordedConsumer
     {
-        public RecordedConsumer(AutorecoveringModel model, string queue)
+        public RecordedConsumer(AutorecoveringModel model, string queue, string consumerTag)
         {
-            Model = model;
-            Queue = queue;
+            if (model == null)
+            {
+                throw new ArgumentNullException(nameof(model));
+            }
+            else
+            {
+                Model = model ?? throw new ArgumentNullException(nameof(model));
```

Comment:
Can `model` ever be `null` (or null-like) here after skiping the section `model == null` ?